app/developers/forms.py

```diff
@@ -46,6 +47,14 @@ class ConditionSelectQuestionForm(FlaskForm):
     def __init__(self, *args, question: "Question", **kwargs):  # type: ignore[no-untyped-def]
         super().__init__(*args, **kwargs)
 
+        self.target_question = question
+
         self.question.choices = [
             (question.id, f"{question.text} ({question.name})") for question in get_supported_form_questions(question)
         ]
+        self.question.validators.append(self.question_depends_on_order)  # type: ignore[attr-defined]
+
+    def question_depends_on_order(self, form: "ConditionSelectQuestionForm", field: "Field") -> None:
```

Comment:
Could use a field validator: https://wtforms.readthedocs.io/en/3.0.x/validators/#custom-validators

but this is fine